src/core/components/hv-route/index.tsx

```diff
@@ -313,6 +323,36 @@ class HvRouteInner extends PureComponent<Types.InnerRouteProps, Types.State> {
   }
 }
 
+/**
+ * Retrieve a nested navigator as a child of the nav-route with the given id
+ */
+const getNestedNavigator = (
+  id?: string,
+  doc?: TypesLegacy.Document,
+): TypesLegacy.Element | undefined => {
+  if (!id || !doc) {
+    return undefined;
+  }
+
+  const routes = doc
+    .getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      TypesLegacy.LOCAL_NAME.NAV_ROUTE,
+    )
+    .filter((n: TypesLegacy.Element) => {
+      return n.getAttribute('id') === id;
+    });
+  const route = routes && routes.length > 0 ? routes[0] : undefined;
+  if (route) {
+    const navigators = route.getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      TypesLegacy.LOCAL_NAME.NAVIGATOR,
+    );
+    return navigators && navigators.length > 0 ? navigators[0] : undefined;
```

Comment:
I assume here, by extracting the first navigator, we want to retrieve the first immediate `<navigator>` descendant of `route`?
if so we might want to restrict the selection to `route.childNodes`?

Reply:
Thanks. I had the same thought and had the fix done but didn't push before I stepped away. It's updated now.